server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -574,7 +576,8 @@ void moveShards() {
 
             ShardRouting shardRouting = it.next();
 
-            if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getShardPool(shardRouting, allocation))) {
+            if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getShardPool(shardRouting, allocation))
+                && !(FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX) && isPartialShard(shardRouting, allocation))) {
```

Comment:
[minor]: move this check to a private method in the class as it is repeated multiple times

Reply:
Also adding this new && condition will impact for remote snapshots for which also REMOTE_CAPABLE will be set.

Reply:
> [minor]: move this check to a private method in the class as it is repeated multiple times

done

> Also adding this new && condition will impact for remote snapshots for which also REMOTE_CAPABLE will be set.

the new condition is && plus a ! , which means the current flow for remote snapshots will not be impacted unless feature flag is enabled for tiered remote index, and even if the feature flag is enabled, the condition will not be satisfied only in case when the shard has index setting with data locality set to PARTIAL as the partial shards are expected to go to the dedicated search nodes.